ee/tabby-webserver/src/service/repository.rs

```diff
@@ -37,6 +41,62 @@ impl RepositoryService for DbConn {
             .await?;
         Ok(true)
     }
+
+    async fn search_files(
+        &self,
+        name: String,
+        pattern: String,
+        top_n: usize,
+    ) -> Result<Vec<FileEntry>> {
+        if pattern.trim().is_empty() {
+            return Ok(vec![]);
+        }
+        let git_url = self.get_repository_by_name(name.clone()).await?.git_url;
+        let config = RepositoryConfig::new_named(name, git_url);
+        let matching = match_pattern(&config.dir(), &pattern, top_n)
+            .await
+            .map_err(anyhow::Error::from)?;
+        Ok(matching)
+    }
+}
+
+async fn match_pattern(
+    base: &Path,
+    pattern: &str,
+    limit: usize,
+) -> Result<Vec<FileEntry>, anyhow::Error> {
+    let mut nucleo = nucleo::Matcher::new(nucleo::Config::DEFAULT.match_paths());
+    let needle = nucleo::pattern::Pattern::new(
+        pattern,
+        nucleo::pattern::CaseMatching::Ignore,
+        nucleo::pattern::Normalization::Smart,
+        nucleo::pattern::AtomKind::Fuzzy,
+    );
+    let mut scored_entries: Vec<(_, _)> = Walk::new(base)
```

Comment:
This is a synchronous file walk, and will stall the server. The matching style looks interesting, and may be an improvement, but this is the weakness of the `ignore` crate that I had been talking about.